Src/AutoFakeItEasy/FakeItEasyMethodQuery.cs

```diff
@@ -74,8 +74,46 @@ public IEnumerable<ParameterInfo> Parameters
 
             public object Invoke(IEnumerable<object> parameters)
             {
-                return new Fake<T>(
-                    b => b.WithArgumentsForConstructor(parameters));
+                var genericFakeType = typeof (Fake<>).MakeGenericType(typeof (T));

```

Comment:
I'm afraid I don't quite follow.
The existing AutoFakeItEasyUnitTest project covers this code (as does the new AutoFakeItEasy.FakeItEasy2.UnitTest project that you were hoping to avoid). This code was written specifically to cause those two test suites to pass.

Granted the tests verify what happens when there's no constructor with length other than 1, or where the parameter isn't an action, or where the action doesn't act on a type that has a `WithArgumentsForConstructor`, but all FakeItEasys after 1.7.4257.42 meet the requirements, and the tests serve to ensure that any FakeItEasy dropped into the tests continues to do so. The same statements could be made about the original tests and code as well.